internal/http/services/thumbnails/thumbnails.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2018-2019 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package thumbnails
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"errors"
+	"image"
+	"image/jpeg"
+	"image/png"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/rhttp"
+	"github.com/nfnt/resize"
+	"github.com/rs/zerolog"
+)
+
+type thumnailhandler struct {
+	Parent  *svc
+	Log     *zerolog.Logger
+	Writer  http.ResponseWriter
+	Request *http.Request
+	URL     string
+
+	FileType string
+	Size     uint
+
+	TempFileName string
+}
+
+func newThumbnailHandler(s *svc, w http.ResponseWriter, r *http.Request) *thumnailhandler {
```

Comment:
functions are package scoped. a more appropriate name would be `newHandler`.

In general I'd move away from postfixing this with "-Handler" as it can be read as "thumbnailHandler implements the Handler interface" [source](https://golang.org/pkg/net/http/#Handler). Instead, your type `thumbnailHandler` is an abstraction that does not implement such interface.

You're implementing the Handler interface on your thumbnails>service.go on `svc` [method set](https://golang.org/ref/spec#Method_sets)